scripts/install_moodle.sh

```diff
@@ -650,9 +662,31 @@ EOF
     # Moodle requirements
     sudo apt-get -y update > /dev/null
     sudo apt-get install -y --force-yes graphviz aspell php-common php-soap php-json php-redis > /tmp/apt6.log
-    sudo apt-get install -y --force-yes php-bcmath php-gd php-xmlrpc php-intl php-xml php-bz2 >> /tmp/apt6.log
+    sudo apt-get install -y --force-yes php-bcmath php-gd php-xmlrpc php-intl php-xml php-bz2 php-pear php-mbstring php-dev mcrypt >> /tmp/apt6.log
     if [ $dbServerType = "mysql" ]; then
         sudo apt-get install -y --force-yes php-mysql
+    elif [ $dbServerType = "mssql" ]; then
+        # Download and build php/mssql driver
+        /usr/bin/curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add -  
+        /usr/bin/curl https://packages.microsoft.com/config/ubuntu/16.04/prod.list > /etc/apt/sources.list.d/mssql-release.list
+        sudo apt-get update
+        sudo ACCEPT_EULA=Y apt-get install msodbcsql mssql-tools unixodbc-dev -y
+        echo 'export PATH="$PATH:/opt/mssql-tools/bin"' >> ~/.bash_profile
+        echo 'export PATH="$PATH:/opt/mssql-tools/bin"' >> ~/.bashrc
+        source ~/.bashrc   
+             
+       #Build mssql driver
+       /usr/bin/pear config-set php_ini `php --ini | grep "Loaded Configuration" | sed -e "s|.*:\s*||"` system
+       /usr/bin/pecl install sqlsrv
+       /usr/bin/pecl install pdo_sqlsrv
+       PHPVER=`/usr/bin/php -r "echo PHP_VERSION;" | /usr/bin/cut -c 1,2,3`
+       echo "extension=sqlsrv.so" >> /etc/php/$PHPVER/fpm/php.ini
+       echo "extension=pdo_sqlsrv.so" >> /etc/php/$PHPVER/fpm/php.ini
+       echo "extension=sqlsrv.so" >> /etc/php/$PHPVER/apache2/php.ini
+       echo "extension=pdo_sqlsrv.so" >> /etc/php/$PHPVER/apache2/php.ini
+       echo "extension=sqlsrv.so" >> /etc/php/$PHPVER/cli/php.ini
+       echo "extension=pdo_sqlsrv.so" >> /etc/php/$PHPVER/cli/php.ini
+        
```

Comment:
These lines (L669-L689) are duplicated in scripts/setup_webserver.sh below. Could you please move these lines to scripts/helper_functions.sh as a shell function and avoid code duplication?

Reply:
fixed